packages/core/src/components/editable-text/editableText.tsx

```diff
@@ -105,6 +105,13 @@ export interface IEditableTextProps extends IntentProps, Props {
      */
     selectAllOnFocus?: boolean;
 
+    /**
+     * Whether the input should stop the event propagation when escape is pressed
```

Comment:
I think it would be helpful to add documentation about the motivation for this feature here, like you did in the PR description.

```suggestion
     * Whether the input should stop the event propagation when escape is pressed.
     * This is useful, for example, when this component is used inside a `<Dialog>` with
     * `canEscapeKeyClose` enabled.
```